core/src/no/ntnu/tdt4240/asteroids/game/effect/BaseEffect.java

```diff
@@ -57,7 +56,7 @@ public void tick(PooledEngine engine, Entity entity, EffectComponent component,
     }
 
     private void setTexture(Entity entity) {
-        if (getEffectTexture() == null) return;
+        if (getEffectTexture() == null || getEffectTexture() == oldRegion) return;
```

Comment:
What is the logic behind || getEffectTexture() == oldRegion ? If the new texture is equal to the backup of the previous one, return ? Not sure I understand